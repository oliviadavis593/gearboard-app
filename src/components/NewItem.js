import React, { Component } from 'react';

class NewItem extends Component {
    render() {
        return(
            <form>
                <h1>New Gear Item</h1>
                <div>
                    <label>Instrument Name</label>
                    <input 
                    type='text'
                    name='instrument-name'
                    placeholder='Fender Stratocaster'
                    required
                    />
                </div>
                <div>
                    <label>Description</label>
                    <textarea 
                    name='description'
                    placeholder='What are the specs and features of your gear?'
                    />
                </div>
                <div>
                    <label>Comments</label>
                    <textarea 
                    name='comments'
                    placeholder='What are the pros and cons of your gear?'
                    />
                </div>
                <button type='submit'>
                    Submit
                </button>
            </form>
        )
    }
}

export default NewItem; 