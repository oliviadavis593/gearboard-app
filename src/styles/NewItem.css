fieldset {
    border: 0;
}

.gb-new-item__form {
    width: 300px; 
    height: 500px; 
    margin: 30px auto;
    background-color: #eee; 
    box-shadow: 0 2px 4px 0 rgba(0,0,0,0.4);
}

legend {
    text-align: center; 
}


#gear-name,
#features, 
#comments,
#rating {
    margin-bottom: 30px;
}

input {
    border: none;
    display: flex; 
    flex-direction: column; 
    width: 200px; 
    height: 25px; 
}

textarea {
    border: none; 
    display: flex; 
    flex-direction: column; 
    width: 200px; 
    height: 45px; 
    resize: none; 
}

.gb-add__button {
    text-decoration: none;
    color: #000;
    font-family: 'Merriweather', serif;
    background-color: #e67e22;
    width: 100px; 
    height: 40px; 
    border: none; 
    font-size: 15px;
    margin-left: 100px; 
    margin-top: 25px; 
    border-radius: 20px; 
}

.gb-add__button:hover{
    color: #fff;
    transition: 0.5s; 
}