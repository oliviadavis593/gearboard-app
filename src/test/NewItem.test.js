import React from 'react';
import ReactDOM from 'react-dom'
import NewItem from '../components/NewItemForm/NewItemForm'


describe('Footer component', () => {
    it(`renders without crashing`, () => {
        const div = document.createElement('div')
        ReactDOM.render(<NewItem />, div)
        ReactDOM.unmountComponentAtNode(div)
    })
})