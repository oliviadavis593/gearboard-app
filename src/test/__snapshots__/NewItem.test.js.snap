// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders the UI as expected 1`] = `
<div>
  <header>
    <div
      className="gb-header__container"
    >
      <header
        className="gb-main__header"
      >
        <a
          aria-current={null}
          href="/homepage"
          onClick={[Function]}
        >
          <h1>
            GearBoard
          </h1>
        </a>
      </header>
      <nav
        className="nav"
      >
        <div
          className="nav__main-links"
        >
          <a
            aria-current={null}
            className="gb-link__all"
            href="/homepage"
            onClick={[Function]}
          >
            All Gear
          </a>
          <a
            aria-current={null}
            className="gb-link__new"
            href="/new-item"
            onClick={[Function]}
          >
            Submit new item
          </a>
          <a
            aria-current="page"
            className="gb-link__logout active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Logout
          </a>
        </div>
        <span
          className="navbar-toggle"
          id="js-navbar-toggle"
        />
      </nav>
    </div>
  </header>
  <main>
    <form
      className="gb-new-item__form"
      onSubmit={[Function]}
    >
      <div
        className="gb-add-item__error"
      />
      <fieldset>
        <legend>
          <h2>
            Add New Gear
          </h2>
        </legend>
        <label
          htmlFor="gear-name"
        >
          Gear Name:
        </label>
        <input
          id="gear-name"
          name="gear_name"
          onChange={[Function]}
          required={true}
          type="text"
        />
        <label
          htmlFor="features"
        >
          Features:
        </label>
        <textarea
          id="features"
          name="features"
          onChange={[Function]}
          required={true}
          type="textarea"
        />
        <label
          htmlFor="comments"
        >
          Comments:
        </label>
        <textarea
          id="comments"
          name="comments"
          onChange={[Function]}
          type="textarea"
        />
        <label
          htmlFor="rating"
        >
          Rating:
        </label>
        <select
          id="gb-rating"
          name="rating"
          onChange={[Function]}
        >
          <option
            value=""
          />
          <option
            value="🎸"
          >
            🎸
          </option>
          <option
            value="🎸🎸"
          >
            🎸🎸
          </option>
          <option
            value="🎸🎸🎸"
          >
            🎸🎸🎸
          </option>
          <option
            value="🎸🎸🎸🎸"
          >
            🎸🎸🎸🎸
          </option>
          <option
            value="🎸🎸🎸🎸🎸"
          >
            🎸🎸🎸🎸🎸
          </option>
        </select>
      </fieldset>
      <button
        className="gb-add__button"
        type="submit"
      >
        Add Gear
      </button>
    </form>
  </main>
</div>
`;
